File: Inventory.ts
export class Inventory {
    private items = new Map<string, number>();

    addItem(itemName: string) {
        this.items.set(itemName, (this.items.get(itemName) || 0) + 1);
    }

    useItem(itemName: string): boolean {
        const count = this.items.get(itemName) || 0;
        if (count > 0) {
            if (count === 1) this.items.delete(itemName);
            else this.items.set(itemName, count - 1);
            return true;
        }
        return false;
    }

    hasItem(itemName: string): boolean {
        return (this.items.get(itemName) || 0) > 0;
    }

    displayInventory(): string {
        let sb = 'Inventory:\n';
        this.items.forEach((count, name) => (sb += `${name}: ${count}\n`));
        return sb;
    }
}

File: Inventory.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.mazegame;

/**
 *
 * @author lsuti
 */
import java.util.*;

public class Inventory {

    private final Map<String, Integer> items = new HashMap<>();

    public void addItem(String name) {
        items.put(name, items.getOrDefault(name, 0) + 1);
    }

    public boolean useItem(String name) {
        if (items.containsKey(name) && items.get(name) > 0) {
            items.put(name, items.get(name) - 1);
            if (items.get(name) == 0) {
                items.remove(name);
            }
            return true;
        }
        return false;
    }

    public boolean hasItem(String name) {
        return items.getOrDefault(name, 0) > 0;
    }

    public String displayInventory() {
        if (items.isEmpty()) {
            return "Inventory Empty";
        }
        StringBuilder sb = new StringBuilder("Inventory:\n");
        items.forEach((k, v) -> sb.append(k).append(": ").append(v).append("\n"));
        return sb.toString();
    }
}
